MainWindow.xaml.cs

```diff
@@ -219,5 +220,47 @@ private void UpdateEditWindows()
                 _edit_metadata_window.Content = user_control;
             }
         }
+
+        #region Test
+        public void TestInputPack(Data.DynamicInputPack inputPack)
+        {
+            foreach (Data.HostDevice device in inputPack.HostDevices)
+            {
+                foreach (Data.HostKey key in device.HostKeys)
+                {
+                    if (!File.Exists(key.TexturePath))
+                    {
+                        key.TexturePath = GetNewImage(key.Name, key.TexturePath);
+                    }
+                }
+            }
+            foreach (Data.DynamicInputTexture texture in inputPack.Textures)
+            {
+                if (!File.Exists(texture.TexturePath))
+                {
+                    texture.TexturePath = GetNewImage(texture.TextureHash, texture.TexturePath);
+                }
+            }
+        }
+
+        private string GetNewImage(string name, string paht)
+        {
+            MessageBoxResult MessageResult;
+            MessageResult = MessageBox.Show(String.Format("'{1}'\nThe image of '{0}' could not be found!\nsearch for the picture?", name, paht), name, MessageBoxButton.YesNo);
+            if (MessageResult == MessageBoxResult.Yes)
```

Comment:
Two issues I have here:

1.   If the user selects "no", they have a broken image but no way to fix after the fact.
2.   If the user had previously gotten the hash from the image, then allowing them to choose any image could technically have any hash.  Admittedly the hash is just part of the filename, so it's not like there was strict error checking before but..

I think 1) we can punt on.  2) however, I think could easily be resolved by searching for an image with the same filename as before.

Reply:
1. I have planned in conjunction with other improvements.
2. good idea, I have added a matching filter.